doc/dev/custom_devices.rst

```diff
@@ -172,3 +178,84 @@ After doing so, Catalyst should be able to interface with your custom device.
     @qml.qnode(CustomDevice(wires=1))
     def f():
         return measure(0)
+
+
+Below is an example configuration file
+
+.. code-block:: toml
```

Comment:
Could we use this draft for the dummy device instead?

``` toml
schema = 1 #  May be useful for schema-relate versioning of this file structure across releases.

[device]
name = "device_name"
version = "0.0.0"
precision = ['float32', 'float64'] #not every device supports FP64, and some may even have FP16, BF16, etc

[operations]
observables = []
[[operations.gates]] # This creates a table structure in TOML
full = [] # list of supported native gates with both "exact-shots" and "finite-shots"
exactshots = [] # list of supported native gates with "exact-shots"
finiteshots = [] # list of supported native gates with "finite-shots"
matrix = [] # list of supported gates via applying matrices, like `OrbitalRotation` in Lightning simulators

[measurements]
exactshots = []  # list of supported measurements with exact-shots
finiteshots = []  # list of supported measurements with finite-shots

[options]
qjit_compatible = false  # True if it's supported by Catalyst
control_flow = false  # True if it has real-time control-flow support
dynamic_qubit_management = false  # True if it supports dynamic qubit management
```

Reply:
I think `DummyDevice` does support observables and gates. It supports all gates supported by lightning runtime (even though the semantics are different, it just does nothing). So I think it is incorrect to say that the device does not support anything.

Reply:
See definition of `DummyDevice` inlined here:

```cpp
struct DummyDevice final : public Catalyst::Runtime::QuantumDevice {
    DummyDevice([[maybe_unused]] const std::string &kwargs) {}
    ~DummyDevice() = default; // LCOV_EXCL_LINE

    DummyDevice &operator=(const QuantumDevice &) = delete;
    DummyDevice(const DummyDevice &) = delete;
    DummyDevice(DummyDevice &&) = delete;
    DummyDevice &operator=(QuantumDevice &&) = delete;

    auto AllocateQubit() -> QubitIdType override { return 0; }
    auto AllocateQubits(size_t num_qubits) -> std::vector<QubitIdType> override
    {
        return std::vector<QubitIdType>(num_qubits);
    }
    [[nodiscard]] auto Zero() const -> Result override { return NULL; }
    [[nodiscard]] auto One() const -> Result override { return NULL; }
    auto Observable(ObsId, const std::vector<std::complex<double>> &,
                    const std::vector<QubitIdType> &) -> ObsIdType override
    {
        return 0;
    }
    auto TensorObservable(const std::vector<ObsIdType> &) -> ObsIdType override { return 0; }
    auto HamiltonianObservable(const std::vector<double> &, const std::vector<ObsIdType> &)
        -> ObsIdType override
    {
        return 0;
    }
    auto Measure(QubitIdType) -> Result override
    {
        bool *ret = (bool *)malloc(sizeof(bool));
        *ret = true;
        return ret;
    }

    void ReleaseQubit(QubitIdType) override {}
    void ReleaseAllQubits() override {}
    [[nodiscard]] auto GetNumQubits() const -> size_t override { return 0; }
    void SetDeviceShots(size_t shots) override {}
    [[nodiscard]] auto GetDeviceShots() const -> size_t override { return 0; }
    void StartTapeRecording() override {}
    void StopTapeRecording() override {}
    void PrintState() override {}
    void NamedOperation(const std::string &, const std::vector<double> &,
                        const std::vector<QubitIdType> &, bool) override
    {
    }

    void MatrixOperation(const std::vector<std::complex<double>> &,
                         const std::vector<QubitIdType> &, bool) override
    {
    }
```